_data-prepper/common-use-cases/s3-logs.md

```diff
@@ -0,0 +1,176 @@
+2. Configure [bucket notifications](https://docs.aws.amazon.com/AmazonS3/latest/userguide/ways-to-add-notification-config-to-bucket.html) for SQS. Use the `s3:ObjectCreated:*` event type.
+3. Grant [AWS IAM](https://docs.aws.amazon.com/IAM/latest/UserGuide/introduction.html) permissions to Data Prepper for accessing SQS and S3.
+4. (Recommended) Create an [SQS dead-letter queue](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-dead-letter-queues.html) (DLQ).
+5. (Recommended) Configure an SQS re-drive policy to move failed messages into the DLQ.
+
+### Setting permissions for Data Prepper
+
+To view S3 logs, Data Prepper needs access to Amazon SQS and S3.
+Use the following example to set up permissions:
+
+```json
+{
+    "Version": "2012-10-17",
+    "Statement": [
+        {
+            "Sid": "s3-access",
+            "Effect": "Allow",
+            "Action": "s3:GetObject",
+            "Resource": "arn:aws:s3:::<YOUR-BUCKET>/*"
+        },
+        {
+            "Sid": "sqs-access",
+            "Effect": "Allow",
+            "Action": [
+                "sqs:DeleteMessage",
+                "sqs:ReceiveMessage"
+            ],
+            "Resource": "arn:aws:sqs:<YOUR-REGION>:<123456789012>:<YOUR-SQS-QUEUE>"
+        },
+        {
+            "Sid": "kms-access",
+            "Effect": "Allow",
+            "Action": "kms:Decrypt",
+            "Resource": "arn:aws:kms:<YOUR-REGION>:<123456789012>:key/<YOUR-KMS-KEY>"
+        }
+    ]
+}
+```
+
+If your S3 objects or SQS queues do not use KMS, you can remove the `kms:Decrypt` permission.
+
+### SQS dead-letter queue
+
+The are two options for how to handle errors resulting from processing S3 objects.
+
+- Use an SQS dead-letter queue (DLQ) to track the failure. This is the recommended approach.
+- Delete the message from SQS. You must manually find the S3 object and correct the error.
+
+The following diagram shows the system architecture when using SQS with DLQ.
+
+<img src="{{site.url}}{{site.baseurl}}/images/data-prepper/s3-source/s3-architecture-dlq.jpg" alt="S3 source architecture with dlq">{: .img-fluid}
+
+To use an SQS dead-letter queue, perform the following steps:
+
+1. Create a new SQS standard queue to act as your DLQ.
+2. Configure your SQS's redrive policy [to use your DLQ](https://docs.aws.amazon.com/AWSSimpleQueueService/latest/SQSDeveloperGuide/sqs-configure-dead-letter-queue.html). Consider using a low value such as 2 or 3 for the "Maximum Receives" setting.
+3. Configure the Data Prepper `s3` source to use `retain_messages` for `on_error`. This is the default behavior.
+
+## Pipeline design
+
+Create a pipeline to read logs from S3, starting with an [`s3`]({{site.url}}{{site.baseurl}}/data-prepper/pipelines/configuration/sources/s3/) source plugin. Use the following example for guidance. 
```

Comment:
"Create a pipeline for reading logs from S3"? Replace the terminal period with a colon.